### PR TITLE
Feature/ab 604 remote test gateway

### PR DESCRIPTION
**Pull request by mgonzales-rs**
Adds remote test calling function to Gateway

### CODE REVIEW

File: eze/cli/commands/test_remote_commands.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+)
+@click.option(
+    "--online",
+    is_flag=True,
+    help="Especify if you want to run the test online instead of locally",
+)
+def test_remote_commands(state, config_file: str, scan_type, url: str, online: bool) -> None:
+    """Eze run scan in a remote repository"""
+    if online:
+        api_url = "https://lff3adj48f.execute-api.eu-west-2.amazonaws.com/staging/v1/api/remote-scan"
```
Comment (Anthony-Mckale) [inline eze/cli/commands/test_remote_commands.py:42]: FIXME: hardcoded url, should be EZE_REMOTE_SCAN_ENDPOINT or something similar as per
C:\dev\repos\eze-cli\eze\plugins\reporters\eze.py

File: eze/cli/commands/test_remote_commands.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+@click.option(
+    "--scan-type",
+    "-s",
+    help="named custom scan type to run aka production can include run type aka 'safety:test-only'",
+    required=False,
+)
+@click.option(
+    "--url",
+    "-u",
+    help="Especify the url of the remote repository to run scan. ex https://user:pass@github.com/repo-url",
```
Comment (Anthony-Mckale) [inline eze/cli/commands/test_remote_commands.py:31]: Spelling ? : Especify

**Review by Anthony-Mckale — CHANGES_REQUESTED** [review]
needs some urls and apikey set in environment variables, might be worth adding flags for the url and apikey too

File: eze/cli/commands/test_remote_commands.py (hunk 1 above)
Comment (mgonzales-rs) [reply eze/cli/commands/test_remote_commands.py:42]: ok, nice

File: eze/cli/commands/test_remote_commands.py (hunk 2 above)
Comment (mgonzales-rs) [reply eze/cli/commands/test_remote_commands.py:31]: :O spanish messing up with me

File: eze/cli/commands/test_remote_commands.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    help="Especify if you want to run the test online instead of locally",
+)
+def test_remote_commands(state, config_file: str, scan_type, url: str, online: bool) -> None:
+    """Eze run scan in a remote repository"""
+    if online:
+        api_url = "https://lff3adj48f.execute-api.eu-west-2.amazonaws.com/staging/v1/api/remote-scan"
+        data = {"remote-url": url}
+        try:
+            req = urllib.request.Request(
+                api_url, data=pretty_print_json(data).encode("utf-8"), headers={"Authorization": "---"}
```
[the remark this answers is not in the document]
Comment (mgonzales-rs) [reply eze/cli/commands/test_remote_commands.py:46]: cool.

**Comment by mgonzales-rs** [comment]
Changes from ab#601 were merged so less commits are applicable here

**Pull request merged**